dotCMS/src/main/java/com/dotcms/contenttype/business/ContentTypeAPIImpl.java

```diff
@@ -137,6 +133,38 @@ public ContentType find(final String inodeOrVar) throws DotSecurityException, Do
     throw new DotSecurityException("User " + user + " does not have READ permissions on ContentType " + type);
   }
 
+  @Override
+  public Optional<List<ContentType>> find(final List<String> varNames, final int offset, final int limit,
+                                          final String orderBy) throws DotSecurityException, DotDataException {
+    if (UtilMethods.isNotSet(varNames)) {
+      return Optional.empty();
+    }
+    final int internalOffset = 0;
+    final int internalLimit = 500;
+    final List<ContentType> contentTypeList;
+    if (offset > 0 || limit > 0) {
+      int adjustedLimit = offset + limit;
+      adjustedLimit = adjustedLimit > varNames.size() ? varNames.size() : limit;
+      final List<String> varNamesSubList = varNames.subList(offset, adjustedLimit);
+      contentTypeList = this.contentTypeFactory.find(varNamesSubList, internalOffset, internalLimit, orderBy);
+    } else {
+      contentTypeList = this.contentTypeFactory.find(varNames, internalOffset, internalLimit, orderBy);
+    }
+    final List<ContentType> accessibleContentTypes = contentTypeList.stream()
+            .filter(type -> {
+              try {
```

Comment:
I would move this to a method and reduce to 
.filter(this.doesUserHaveReadPermission(type, user)